install_scripts/templates/kubernetes/init.sh

```diff
@@ -458,9 +458,27 @@ registryDeploy() {
     done
     REGISTRY_ADDRESS_OVERRIDE="$registryIP:5000"
     YAML_GENERATE_OPTS="$YAML_GENERATE_OPTS registry-address-override=$REGISTRY_ADDRESS_OVERRIDE"
+
     addInsecureRegistry "$SERVICE_CIDR"
-    waitForRegistry
+    # check if there are worker nodes that need to be configured for the insecure registry
+    local workers=$(kubectl get nodes --selector='!node-role.kubernetes.io/master' -o jsonpath='{.items[*].metadata.name}')
+    local numMasters=$(kubectl get nodes --selector='node-role.kubernetes.io/master' | sed '1d' | wc -l)
+    # check the ADDED_INSECURE_REGISTRY flag and the number of masters to ensure this is only shown once
+    if [ "$ADDED_INSECURE_REGISTRY" = "1" ] && [ -n "$workers" ] && [ "$numMasters" -eq "1" ]; then
+        cat <<EOF
+Configure Docker on all worker nodes to use http when pulling from the in-cluster registry before proceeding.
```

Comment:
is this automatic if you run the node join script? if not can it be?

Reply:
Yes, it's automatic in the node join script.